### PR TITLE
Improve configuration docs

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/guide/en/widget-configuring.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Configuring the widget
 
+Widgets could be configured on multiple levels.
+
+- Configuration passed to the `widget()` method call;
+- Widget factory configuration:
+  - With definitions;
+  - With themes;
+- With a theme specified at the `widget()` method call.
```
Comment (vjik) [inline docs/guide/en/widget-configuring.md:9]:
```suggestion
Widget configuration is compiled from multiple parts:

- configuration passed to the `widget()` method call;
- extra configuration from the widget itself by specified theme;
- configuration defined in widget factory configuration:
  - in definitions;
  - in themes.
```

File: docs/guide/en/widget-configuring.md [hunk 2]
```diff
@@ -44,3 +57,67 @@ parameter:
...
+                    'color' => 'black',
+                ],
+            ],
+        ],
+    ],
+    defaultTheme: 'black',
+);
+```
+
+## With a theme specified at the `widget()` method call.
```
Comment (vjik) [inline docs/guide/en/widget-configuring.md:116]: Need info about extra configuration from widget itself.

**Review by vjik — CHANGES_REQUESTED** [review]

File: docs/guide/en/widget-configuring.md (hunk 1 above)
Comment (samdark) [reply docs/guide/en/widget-configuring.md:9]: I'd use "combined", not "compiled".

File: docs/guide/en/widget-configuring.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+Widgets could be configured on multiple levels.
+
+- Configuration passed to the `widget()` method call;
+- Widget factory configuration:
+  - With definitions;
+  - With themes;
+- With a theme specified at the `widget()` method call.
+
+More specific configuration has more priority. Configuration passed to the `widget()` method call has more priority 
+than configuration through widget factory.
```
Comment (samdark) [inline docs/guide/en/widget-configuring.md:12]:
```suggestion
More specific configuration has more priority. For example, configuration passed to the `widget()` method call has more priority 
than configuration through widget factory.
```

File: docs/guide/en/widget-configuring.md (hunk 2 above)
Comment (samdark) [reply docs/guide/en/widget-configuring.md:116]: Do you mean the fact that theme config is merged with widget call level config?

Comment (vjik) [reply docs/guide/en/widget-configuring.md:116]: No. I means such cases as field in `yiisoft/form`: https://github.com/yiisoft/form/blob/e0daafe346b4ac20ed4893451e9a361c47bc9cf0/src/Field/Base/BaseField.php#L150

We have four levels:

1) configuration passed to the `widget()` method call;
2) extra configuration from the widget itself by specified theme (see case in `yiisoft/form`);
3) configuration defined in widget factory themes:
4) configuration defined in widget factory definitions.

**Review by samdark — APPROVED** [review]

**Review by vjik — APPROVED** [review]

**Pull request merged**